src/tests/test_contracts/test_yas_pool.cairo

```diff
@@ -29,22 +31,178 @@ mod YASPoolTests {
+        fn test_fails_if_price_is_min_sqrt_ratio_minus_1() {
+            let yas_pool = deploy(
+                FACTORY_ADDRESS(), TOKEN_A(), TOKEN_B(), 5, IntegerTrait::<i32>::new(1, false)
+            );
+
+            let sqrt_price_X96 = FixedTrait::new(MIN_SQRT_RATIO - 1, false);
+            yas_pool.initialize(sqrt_price_X96);
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        #[should_panic(expected: ('R', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_if_price_is_too_high() {
+            let yas_pool = deploy(
+                FACTORY_ADDRESS(), TOKEN_A(), TOKEN_B(), 5, IntegerTrait::<i32>::new(1, false)
+            );
+
+            let sqrt_price_X96 = FixedTrait::new(pow(2, 160) - 1, false);
+            yas_pool.initialize(sqrt_price_X96);
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        #[should_panic(expected: ('R', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_if_price_is_max_sqrt_ratio() {
+            let yas_pool = deploy(
+                FACTORY_ADDRESS(), TOKEN_A(), TOKEN_B(), 5, IntegerTrait::<i32>::new(1, false)
+            );
+
+            let sqrt_price_X96 = FixedTrait::new(MAX_SQRT_RATIO, false);
+            yas_pool.initialize(sqrt_price_X96);
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        fn test_can_be_initialized_at_min_sqrt_ratio() {
+            let mut state = STATE();
+
+            let sqrt_price_X96 = FixedTrait::new(MIN_SQRT_RATIO, false);
+            YASPoolImpl::initialize(ref state, sqrt_price_X96);
+
+            let expected = Slot0 {
+                sqrt_price_X96: FixedTrait::new(MIN_SQRT_RATIO, false),
+                tick: min_tick(IntegerTrait::<i32>::new(1, false)),
+                fee_protocol: 0,
+                unlocked: true
+            };
+
+            assert(InternalImpl::get_slot_0(@state) == expected, 'slot 0 wrong initialization');
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        fn test_can_be_initialized_at_max_sqrt_ratio_minus_1() {
+            let mut state = STATE();
+
+            let sqrt_price_X96 = FP64x96Impl::new(MAX_SQRT_RATIO - 1, false);
+            YASPoolImpl::initialize(ref state, sqrt_price_X96);
+
+            let expected = Slot0 {
+                sqrt_price_X96: sqrt_price_X96,
+                tick: max_tick(IntegerTrait::<i32>::new(1, false))
+                    - IntegerTrait::<i32>::new(1, false),
+                fee_protocol: 0,
+                unlocked: true
+            };
+
+            assert(InternalImpl::get_slot_0(@state) == expected, 'slot 0 wrong initialization')
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        fn test_sets_initial_variables() {
+            let mut state = STATE();
+
+            // sqrt_price_X96 is the result of encode_price_sqrt(1, 2) on v3-core typescript impl. 
+            let sqrt_price_X96 = FP64x96Impl::new(56022770974786139918731938227 - 1, false);
```

Comment:
Why are you using `56022770974786139918731938227 - 1` instead of just `56022770974786139918731938227`?

Reply:
Solved in the latest commit